UnitTests/GitCommandsTests/Git/ExecutableExtensionsTests.cs

```diff
@@ -9,11 +17,27 @@ namespace GitCommandsTests.Git
     public sealed class ExecutableExtensionsTests
     {
         private MockExecutable _executable;
+        private Executable _gitExecutable;
+        private string _appPath;
 
         [SetUp]
         public void SetUp()
         {
             _executable = new MockExecutable();
+
+            // Work around: When running unittest, Application.UserAppDataPath always points to
+            // %APPDATA%Roaming\Microsoft Corporation\Microsoft.TestHost.x86
+            // We need to correct it to %APPDATA%\GitExtensions\GitExtensions for v3 at least
+            var userAppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            var settingPath = Path.Combine(userAppDataPath, "GitExtensions\\GitExtensions\\GitExtensions.settings");
+            var settingContainer = new RepoDistSettings(null, GitExtSettingsCache.FromCache(settingPath), SettingLevel.Unknown);
+            _appPath = Path.Combine(settingContainer.GetString("gitbindir", ""), "git.exe");
+
+            // Execute process in GitExtension working directory, so that git will return success exit-code
+            // git always return non-zero exit code when run git reset outside of git repository
+            // NUnit working directory always default to MS test host
+            var workingDir = Path.GetDirectoryName(Assembly.GetAssembly(typeof(ExecutableExtensionsTests)).Location);
+            _gitExecutable = new Executable(_appPath, workingDir);
```

Comment:
Why do these tests need the actual `git.exe` path? I think this should be mocked out, making all the env var stuff redundant.

Reply:
After discussion with @mstv, I think testing with actual `git.exe` is valid, because we want to see the `Win32Exception` when command line reaches actual threshold. If we don't test this, we would not be able to see differences of max command line length for each executable. For example, `cmd.exe` has a limit of `8191` characters, while `sh.exe` is limited at `32710` instead of `32767` characters. So testing againt `git.exe` would ensure we have correct understanding of `git` command line max length on Windows.